km/km_filesys.c

```diff
@@ -1864,32 +1872,71 @@ size_t km_fs_core_notes_length()
                ret += sizeof(km_nt_event_t);
             }
          } else if (file->sockinfo == NULL) {
+            queuedbytes = 0;
+            if (file->how == KM_FILE_HOW_PIPE_1) {
+               // We are looking at the write end of a pipe, find out how much data is queued
+               queuedbytes = ioctlfionread(file->ofd);
+            }
             ret += km_note_header_size(KM_NT_NAME) + sizeof(km_nt_file_t) +
-                   km_nt_file_padded_size(file->name);
+                   km_nt_file_padded_size(file->name) + km_nt_chunk_roundup(queuedbytes);
          } else {
+            queuedbytes = 0;
+            if (file->how == KM_FILE_HOW_SOCKETPAIR0 || file->how == KM_FILE_HOW_SOCKETPAIR1) {
+               queuedbytes = ioctlfionread(file->ofd);
+            }
             ret += km_note_header_size(KM_NT_NAME) + sizeof(km_nt_socket_t) +
-                   km_nt_file_padded_size(file->name);
+                   km_nt_file_padded_size(file->name) + km_nt_chunk_roundup(queuedbytes);
          }
       }
    }
    return ret;
 }
 
+static inline size_t
+fs_core_save_pipe_contents(char* buf, size_t length, km_file_t* file, int writefd, size_t queuedbytes)
```

Comment:
maybe mention it applies to sockets too?

Reply:
Added a comment describing the interface this function provides.